crypto/cmp/cmp_client.c

```diff
@@ -712,23 +725,8 @@ X509 *OSSL_CMP_exec_certreq(OSSL_CMP_CTX *ctx, int req_type,
         ERR_raise(ERR_LIB_CMP, CMP_R_NULL_ARGUMENT);
         return NULL;
     }
-    if (is_p10 && crm != NULL) {
-        ERR_raise(ERR_LIB_CMP, CMP_R_INVALID_ARGS);
-        return NULL;
-    }
```

Comment:
Is this check not needed?

Reply:
This check has moved to `cmp_msg.c`:
```
    if (type == OSSL_CMP_PKIBODY_P10CR && crm != NULL) {
        ERR_raise(ERR_LIB_CMP, CMP_R_INVALID_ARGS);
        return NULL;
    }
```

Reply:
I've seen that but it was not obvious to me whether it is truly equivalent as it is done at a very different place. OK, anyway.

Reply:
Good anyway that you questioned this.
`OSSL_CMP_exec_certreq()` calls `initial_certreq()` basically right away,
which calls `ossl_cmp_certreq_new()`, which does the same check
(where `is_p10 = req_type == OSSL_CMP_PKIBODY_P10CR`).